packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx

```diff
@@ -71,42 +71,62 @@ const ActiveVisitsTable = () => {
   const currentPathName = window.location.pathname;
   const fromPage = getOriginFromPathName(currentPathName);
 
-  const headerData = useMemo(
-    () => [
-      {
-        id: 0,
-        header: t('visitStartTime', 'Visit Time'),
-        key: 'visitStartTime',
-      },
-      {
-        id: 1,
-        header: t('idNumber', 'ID Number'),
-        key: 'idNumber',
-      },
-      {
-        id: 2,
-        header: t('name', 'Name'),
-        key: 'name',
-      },
-      {
-        id: 3,
-        header: t('gender', 'Gender'),
-        key: 'gender',
-      },
-      {
-        id: 4,
-        header: t('age', 'Age'),
-        key: 'age',
-      },
-      {
-        id: 5,
-        header: t('visitType', 'Visit Type'),
-        key: 'visitType',
-      },
-    ],
-    [t],
+  const computeHeaderData = useCallback(
+    (t, config) => {
+      const headers = [
+        {
+          id: 0,
+          header: t('visitStartTime', 'Visit Time'),
+          key: 'visitStartTime',
+        },
+      ];
+
+      config?.activeVisits?.identifiers?.map((identifier, index) => {
+        headers.push({
+          id: index + 1,
+          header: t(identifier?.header?.key, identifier?.header?.default),
+          key: identifier?.header?.key,
+        });
+      });
+
+      config?.activeVisits?.attributes?.map((attribute, index) => {
+        headers.push({
+          id: index + config?.activeVisits?.identifiers.length + 1,
+          header: t(attribute?.header?.key, attribute?.header?.default),
+          key: attribute?.header?.key,
+        });
+      });
+
+      headers.push(
+        {
+          id: headers[headers.length - 1].id + 2,
```

Comment:
Can we just have one variable tracking the current index for all of these instead of three different index calculations?